### PR TITLE
Serving CGI's correctly -> Parsing of CGI is done correctly and error handling is implemented

### PR DESCRIPTION
**Pull request by nyzss**
Might need further testing with the error handling

### CODE REVIEW

File: src/multiplexer/EventHandler_cgi.cpp [hunk 1]
```diff
@@ -3,54 +3,72 @@
...
 	}
 	if (pid == 0) {
 		setPipe(fd, STDOUT_FILENO);
-		char **args = new char*;
-		args[0] = const_cast<char*>(fname.c_str());
-		execve(fname.c_str(),args, environ);
-		delete args;
+		char **args = new char*[3];
+		args[0] = const_cast<char *>(arguments[0].c_str());
+		args[1] = const_cast<char *>(arguments[1].c_str());
```
Comment (BHTho) [inline src/multiplexer/EventHandler_cgi.cpp:46]: might be good to think of a way around the const_cast later on. It's a little hacky

File: src/multiplexer/EventHandler_cgi.cpp [hunk 2]
```diff
@@ -3,54 +3,72 @@
...
+		{
+			addToEpoll(fd[0]);
+			_cgiManager.addCgiProcess(clientFd, fd[0], pid);
+			_openConns[fd[0]] = EP_CGI;
+		}
+		else
+		{
+			close(fd[0]);
+			return false;
+		}
```
Comment (BHTho) [inline src/multiplexer/EventHandler_cgi.cpp:70]: Nice addition waiting for the pid.

**Review by BHTho — APPROVED** [review]
very nice

File: src/multiplexer/EventHandler_cgi.cpp (hunk 1 above)
Comment (nyzss) [reply src/multiplexer/EventHandler_cgi.cpp:46]: we can strdup it, but that's a whole other headache so might look into it later on

**Pull request merged**